src/Catalog.API/Program.cs

```diff
@@ -1,9 +1,18 @@
 ﻿using Asp.Versioning.Builder;
+using System.Reflection;
 
 var builder = WebApplication.CreateBuilder(args);
 
-builder.AddServiceDefaults();
-builder.AddApplicationServices();
+if (Assembly.GetEntryAssembly()?.GetName().Name == "GetDocument.Insider")
```

Comment:
Instead of putting this in `Program.cs` should we stick it in a helper method in the service defaults project?

If not, maybe we can stick this check into a nicer method name like `IsBuildTimeOpenApi`? Here's what I need in our benchmark apps at https://github.com/aspnet/Benchmarks/pull/2011.